.github/workflows/deploy.yaml

```diff
@@ -0,0 +1,62 @@
+name: aws-deploy
+
+on:
+  push:
+    branches: [ '*' ]
+    tags: [ 'v[0-9]+\.[0-9]+\.[0-9]+' ]
+  pull_request:
+    branches: [ '*' ]
+
+env:
+  REPO_NAME: ${{ github.event.repository.name }}
+  BRANCH_OR_TAG: ${{ github.ref_name }}
+
+concurrency:
+  group: ${{ github.workflow }}
+
+jobs:
+  pre-commit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
+      - uses: pre-commit/action@v3.0.0
+  jinja-lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
+      - name: Install jinja libraries
+        run: |
+          python -m pip install --upgrade jinja2==3.0.1
+      - name: Install jinja linter tool
+        run: git clone --single-branch https://github.com/drm/jinja2-lint.git /tmp/jinja2-lint && pushd /tmp/jinja2-lint && git checkout 75dcd5a
+      - name: Execute jinja linter
+        run: find . -type f -name "*.j2" -exec /tmp/jinja2-lint/j2lint.py '{}' +
+  deploy:
+    name: Upload to Amazon S3
+    runs-on: ubuntu-latest
+    if: ${{ github.ref == 'refs/heads/main' || startsWith(github.ref, 'refs/tags/v') }}
+    needs: [ "pre-commit", "jinja-lint" ]
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+    - name: Assume AWS Role
+      uses: aws-actions/configure-aws-credentials@v1
+      with:
+        aws-region: us-east-1
+        role-to-assume: arn:aws:iam::745159704268:role/sagebase-github-oidc-sage-bio-ProviderRoleawsinfra-17DRC23HA177D
+        role-session-name: GitHubActions-${{ github.repository_owner }}-${{ github.event.repository.name }}-${{ github.run_id }}
+        role-duration-seconds: 120
+    - name: Copy files with the AWS CLI
+      run: |
+        aws s3 cp templates s3://bootstrap-awss3cloudformationbucket-19qromfd235z9/$REPO_NAME/$BRANCH_OR_TAG/ --recursive
```

Comment:
`REPO_NAME` variable is defined in .travis.yml file but not in this file.  Also where does `$BRANCH_OR_TAG` variable come from?

Maybe you can change this to use one of the S3 github actions from the marketplace?

Reply:
REPO_NAME and BRANCH_OR_TAG are defined in the 'env' section. Will do on S3 GHA.

Reply:
ahh ok, sorry i don't know why i missed that. you can just use them directly here instead of setting them in `env`.  also it might be better to switch this to a GH action on a follow on PR.

Reply:
Removed env vars and used GH pseudovars directly, holding in GHA for s3 cp.